akka-projection-core/src/main/scala/akka/projection/internal/HandlerRecoveryImpl.scala

```diff
@@ -45,6 +66,16 @@ import akka.projection.HandlerRecoveryStrategy
       }
     }
 
+    def retryFutureCallback(attemptCount: AtomicInteger): () => Future[Done] = { () =>
+      tryFutureCallback().recoverWith {
+        // using recoverWith instead of `.failed.foreach` to make sure that calls to statusObserver
+        // are invoked in sequential order
```

Comment:
Interesting.  Would it make sense to log our own errors the same way (i.e. logging in line 120 and other places)?